python/pycylon/data/table.pyx

```diff
@@ -198,38 +206,110 @@ cdef class Table:
         s = Status(status.get_code(), b"", -1)
         return s
 
-    def join(self, ctx: CylonContext, table: Table, join_type: str, algorithm: str, left_col: int,
-             right_col: int) -> Table:
+    def _is_pycylon_table(self, obj):
+        return isinstance(obj, Table)
+
+    def _resolve_column_indices_from_column_names(self, column_names: List[
+        str], op_column_names: List[str]) -> List[int]:
+        resolve_col_ids = []
+        for op_col_id, op_column_name in enumerate(op_column_names):
+            for col_id, column_name in enumerate(column_names):
+                if op_column_name == column_name:
+                    resolve_col_ids.append(col_id)
+        return resolve_col_ids
+
+    def _get_join_column_indices(self, table: Table, **kwargs):
+        ## Check if Passed values are based on left and right column names or indices
+        left_cols = kwargs.get('left_on')
+        right_cols = kwargs.get('right_on')
+        column_names = kwargs.get('on')
+
+        table_col_names_list = [self.column_names, table.column_names]
+
+        if left_cols and right_cols and isinstance(left_cols, List) and isinstance(right_cols,
+                                                                                   List):
+            if isinstance(left_cols[0], str) and isinstance(right_cols[0], str):
+                left_cols = self._resolve_column_indices_from_column_names(self.column_names,
+                                                                           left_cols)
+                right_cols = self._resolve_column_indices_from_column_names(table.column_names,
+                                                                            right_cols)
+                self._check_column_names_viable(left_cols, right_cols)
```

Comment:
@nirandaperera @supunkamburugamuve 
the check is called before the values are returned to join or distjoin